### DISCUSSION TITLE
multiple select controls for custom post and taxonomies

### DISCUSSION THREAD

**Discussion by salmanrazak** [elementor/elementor#19545] [Developers Community]
my first select conrol showing all the custom posts types. based on selecting any one second select control show custom post taxonomies.
where 'songs' is my custom posts and song_type is taxonomy.

`$args = array(
			'public'   => true,
			'_builtin' => false
		);
        $output = 'names'; // 'names' or 'objects' (default: 'names')
		$operator = 'and'; // 'and' or 'or' (default: 'and')
        $post_types = get_post_types( $args, $output, $operator );
        $post_type_name = [];
        

        foreach ( $post_types as $post_type => $val ) {
            $post_type_name[$post_type] = $val;
        }
        
        //var_dump($post_type_name);
        
        $this->add_control(
			'post_type',
			[
				'label' => esc_html__( 'Choose a post type', 'plugin-name' ),
				'type' => \Elementor\Controls_Manager::SELECT,
				//'default' => 'solid',
				'options' => $post_type_name,
                'classes' => 'post_types',
			]
		);

        $this->add_control(
            'unique-control-name',
            [
                'label' => esc_html__( 'Control Label', 'plugin-name' ),
                'type' => \Elementor\Controls_Manager::SELECT,
                'condition' => [
                    'post_type' => 'songs',
                ],
            ]
        );`

any advice, help, correction will be appreciated.
![screenshot-localhost-2022 08 21-02_48_06](https://user-images.githubusercontent.com/3185372/185767112-52baeec9-d9b2-4a52-bc36-9e4174223236.png)

**Comment by WebsiteDons** [comment]
Is there an error you are having? If so, what is it? Is the second select supposed to show options based on the selection of the first? If so there is no `options` index set. If that is your inquiry, you'll need to handle that process via AJAX which will get the `onchange` value and send the request to the database or a JSON formatted file. Then return the relative array that matches the top selected.

The `condition` index only determines that the control is shown when the results match.
